lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -1875,7 +1828,9 @@ Scope * ByteCodeGenerator::FindScopeForSym(Scope *symScope, Scope *scope, Js::Pr
 {
     for (scope = scope ? scope->GetEnclosingScope() : currentScope; scope; scope = scope->GetEnclosingScope())
     {
-        if (scope->GetFunc() != funcInfo && scope->GetMustInstantiate() && scope != this->globalScope)
+        if (scope->GetFunc() != funcInfo
+            && (scope->GetMustInstantiate() || (scope->GetScopeType() == ScopeType_Parameter && !scope->GetCanMergeWithBodyScope()))
```

Comment:
As the param scope is in the stack with body scope marking param scope with must intantiate flag seems to cause issues with the environment index calculation.